### PR TITLE
core: fix cancel order and match status updating

### PR DESCRIPTION
**Pull request by buck54321**
Adds a `DB` method `UpdateOrderStatus`. Updates the order status for both orders in `(*Core).negotiate`.

### CODE REVIEW

File: client/db/interface.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ type DB interface {
 	MarketOrders(dex string, base, quote uint32, n int, since uint64) ([]*MetaOrder, error)
 	// SetChangeCoin stores the change coin for the order.
 	SetChangeCoin(order.OrderID, order.CoinID) error
+	// SetChangeCoin sets the order status for an order.
+	UpdateOrderStatus(oid order.OrderID, status order.OrderStatus) error
```
Comment (chappjc) [inline client/db/interface.go:61]: SetChangeCoin -> UpdateOrderStatus

File: client/core/trade.go [hunk 2]
```diff
@@ -288,15 +287,12 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
...
-		err := t.db.UpdateMatch(&match.MetaMatch)
-		if err != nil {
-			return err
-		}
 		// First check that this isn't a match on its own cancel order. I'm not crazy
 		// about this, but I am detecting this case right now based on the Address
 		// field being an empty string.
 		isCancel := t.cancel != nil && msgMatch.Address == ""
 		if isCancel {
+			match.setStatus(order.MatchComplete)
```
Comment (chappjc) [inline client/core/trade.go:295]: Was thinking about this while working on https://github.com/decred/dcrdex/pull/319.  Glad my client/core thinking was right here.

File: client/core/trade.go [hunk 3]
```diff
@@ -567,7 +569,7 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 	for _, coinID := range coinIDs {
 		coin, found := t.coins[hex.EncodeToString(coinID)]
 		if !found {
-			return errs.add("coin ID %x not found", coinID)
+			return errs.add("coin ID %s not found", coinID)
```
Comment (chappjc) [inline client/core/trade.go:572]: I know this is a random fix, but how would you feel about the following instead:

```go
assetID := t.wallets.fromAsset.ID
coinStr, err := asset.DecodeCoinID(assetID, coinID)
	if err != nil {
	log.Warnf("invalid coin for asset %d: %v", assetID, err)
	coinStr = coinID.String()
}
return errs.add("coin ID %s not found", coinStr)
```

The undecoded coinID bytes just bug me.

File: client/core/trade.go [hunk 4]
```diff
@@ -674,16 +677,6 @@ func (t *trackedTrade) redeemMatches(matches []*matchTracker) error {
 		)
 	}
 
-	coinToStr := func(assetID uint32, coinID []byte) string {
-		coinStr, err := asset.DecodeCoinID(assetID, coinID)
-		if err != nil {
-			// Coin IDs come from wallet.Redeem so this shouldn't happen. Warn.
-			log.Warnf("redeemMatches: invalid redeemed coin ID %v: %v", coinID, err)
```
Comment (buck54321) [inline client/core/trade.go:681]: Noting that by using a package function, we do lose the function name in the warning string.

Comment (chappjc) [reply client/core/trade.go:681]: That's why I duped the code, but it's OK.

**Pull request merged**